MariaDB/server
# MDEV-12182 Add client tcp port to audit plugin logs

- [x] *The Jira issue number for this PR is: MDEV-12182*


## Description
In environments with load balancers or proxies, the audit plugin logs
only the IP address, making it difficult to differentiate individual client
connections from the same IP.

Add a new 'port' field to the appropriate event objects to capture the
client's TCP port number. Populate the port field with thd->port  in the
appropriate functions. The audit plugin receives and logs this port field
along  with  other connection information, enabling better identification
of individual client connections.

## Release Notes
N/A

## How can this PR be tested?
- All existing tests pass when running ./mtr. 
- Execute `server_audit.test` file to verify to port logging feature.


## Basing the PR against the correct MariaDB version

## PR quality check

All new code of the whole pull request, including one or several files that are either new files or modified ones, are contributed under the BSD-new license. I am contributing on behalf of my employer Amazon Web Services, Inc.
